File: Java/Compare-Strings.java
public class Solution {
    /**
     * @param A : A string includes Upper Case letters
     * @param B : A string includes Upper Case letter
     * @return :  if string A contains all of the characters in B return true else return false
     * Time complexity: O(N)
     */
    public boolean compareStrings(String A, String B) {
        // write your code here
        int []vis = new int[26];
        for (int i = 0; i < A.length(); i ++) {
        	int temp = (int)(A.charAt(i) - 'A');
        	vis[temp] ++;
        }
        for (int i = 0; i < B.length(); i ++) {
        	int temp = (int)(B.charAt(i) - 'A');
        	if (vis[temp] == 0) return false;
        	vis[temp] --;
        }
        return true;
    }
}

File: Python/Compare-Strings.py
class Solution:
    """
    @param A : A string includes Upper Case letters
    @param B : A string includes Upper Case letters
    @return :  if string A contains all of the characters in B return True else return False
    """
    def compareStrings(self, A, B):
        # write your code here
        if B is None or len(B) == 0:
            return True
        vis = [0] * 256
        for i in A:
            vis[ord(i)] += 1
        for i in B:
            vis[ord(i)] -= 1
            if vis[ord(i)] < 0:
                return False
        return True